pkg/server/operands_handler.go

```diff
@@ -0,0 +1,73 @@
+package server
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/openshift/console/pkg/serverutils"
+	"github.com/operator-framework/kubectl-operator/pkg/action"
+	"k8s.io/klog"
+)
+
+type OperatorMeta struct {
+	Name      string
+	Namespace string
+}
+
+func operandsListHandler(w http.ResponseWriter, r *http.Request) {
+	if r.Method != "GET" {
+		w.Header().Set("Allow", "GET")
+		serverutils.SendResponse(w, http.StatusMethodNotAllowed, serverutils.ApiError{Err: "Method unsupported, the only supported methods is GET"})
+		return
+	}
+	operatorMeta := parseOperatorNameAndNamespace(r.URL.Path)
+
+	cfg := &action.Configuration{}
+	err := cfg.Load()
+	if err != nil {
+		errMsg := fmt.Sprintf("Failed to set load operator client: %v", err)
+		klog.Error(errMsg)
+		serverutils.SendResponse(w, http.StatusInternalServerError, serverutils.ApiError{Err: errMsg})
+	}
+	// We need to manually set the namespace here since the kubectl-operator
+	// is setting the namespace to "default" upon calling the `cfg.Load()` method
+	if len(operatorMeta.Namespace) != 0 {
+		cfg.Namespace = operatorMeta.Namespace
+	}
+
+	operatorListOperands := action.NewOperatorListOperands(cfg)
+
+	operandsList, err := operatorListOperands.Run(context.TODO(), operatorMeta.Name)
+	if err != nil {
```

Comment:
Looks like if any error is returned from the command it is treated the same. In https://github.com/operator-framework/kubectl-operator/pull/48 we introduce a new `OperandListError` that helps differentiate between different error types. It should help external callers.

Reply:
@exdx thanks for the link. Question is if it make sense to surface these error msgs to the user?

Could be confusing since the RC for the request is `400`, at least for:
`CSV underlying operator is not in a succeeded state: custom resource list may not be accurate`

Reply:
Scratch that, I see we already are doing that.